src/Compilers/CSharp/Portable/BoundTree/BoundNodes.xml

```diff
@@ -2083,6 +2109,21 @@
     <Field Name="IsExplicitNotNullTest" Type="bool"/>
   </Node>
 
+  <Node Name="BoundListPatternClause" Base="BoundNode">
+    <Field Name="ElementType" Type="TypeSymbol"/>
+    <Field Name="Subpatterns" Type="ImmutableArray&lt;BoundPattern&gt;" Null="disallow"/>
+    <Field Name="HasSlice" Type="bool"/>
+    <!-- This will be null if input type is an array -->
+    <Field Name="IndexerInfo" Type="IndexerArgumentInfo?" />
+  </Node>
+
+  <Node Name="BoundSlicePattern" Base="BoundPattern">
+    <Field Name="SliceType" Type="TypeSymbol"/>
+    <Field Name="Pattern" Type="BoundPattern?"/>
+    <!-- This will be null if input type is an array -->
```

Comment:
What's the benefit of treating arrays differently? Would it be simpler to store the info/property here in all cases?

Reply:
Array indexer doesn't have a symbol.